### PR TITLE
Var mutation_icon_state now handles all sprite logic and not mutation_type

### PR DESCRIPTION
**Pull request by aceroyuno**
## About The Pull Request
Adds var/mutation_icon_state to the xenomorph base type. All sprite logic is handled by this var as opposed to the mutation_type var. Strains will no longer need an updated sprite sheet and can now use the base sprite with the original .dmi file.

## Why It's Good For The Game
Allows for creation of strains gameplay wise not to be affected by the strains icon logic. And it makes it simpler for checking code for spriters. If it has no sprite for it's strain, it defaults to using the normal icon state.

## Changelog

:cl: aceroy
refactor: Changed xenomorph icon logic to use new var mutation_icon_state and not mutation_type
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm [hunk 1]
```diff
@@ -351,7 +352,7 @@
 
 	mutators.xeno = src
 
-	update_icon_source()
+	update_icon_source() //I'm not sure why this is here. Line 379 calls recalculate everything which calls this.
```
Comment (fira) [inline code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:355]: please remove this comment or word it differently

line number will change as the file is modified in the future

**Review by fira — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm (hunk 1 above)
Comment (aceroyuno) [reply code/modules/mob/living/carbon/xenomorph/Xenomorph.dm]: reworded comment to better explain what is happening

File: code/modules/mob/living/carbon/xenomorph/update_icons.dm [hunk 2]
```diff
@@ -52,14 +52,23 @@
 		return
 
 	if(stat == DEAD)
-		icon_state = "[mutation_type] [caste.caste_type] Dead"
+		icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Dead"
+		if(!icon state) in icon_states(icon_xeno)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/update_icons.dm:56]:
```suggestion
		if(!icon_state) in icon_states(icon_xeno)
```

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/update_icons.dm (hunk 2 above)
Comment (aceroyuno) [reply code/modules/mob/living/carbon/xenomorph/update_icons.dm]: I had to go into VSC to figure out what I was looking at.

File: code/modules/mob/living/carbon/xenomorph/update_icons.dm [hunk 3]
```diff
@@ -52,14 +52,23 @@
 		return
 
 	if(stat == DEAD)
-		icon_state = "[mutation_type] [caste.caste_type] Dead"
+		icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Dead"
+		if(!icon_state) in icon_states(icon_xeno)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/update_icons.dm:56]: what is this trying to do

```
	if(stat == DEAD)
		icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Dead"
		if(!(icon_state in icon_states(icon_xeno)))
			icon_state = "Normal [caste.caste_type] Dead"
	else if(lying)
		if((resting || sleeping) && (!knocked_down && !knocked_out && health > 0))
			icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Sleeping"
			if(!(icon_state in icon_states(icon_xeno)))
				icon_state = "Normal [caste.caste_type] Sleeping"
		else
			icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Knocked Down"
			if(!(icon_state in icon_states(icon_xeno)))
				icon_state = "Normal [caste.caste_type] Knocked Down"
	else
		icon_state = "[mutation_icon_state || mutation_type] [caste.caste_type] Running"
		if(!(icon_state in icon_states(icon_xeno)))
			icon_state = "Normal [caste.caste_type] Running"
```


this?

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/update_icons.dm (hunk 3 above)
Comment (aceroyuno) [reply code/modules/mob/living/carbon/xenomorph/update_icons.dm:56]: Check to see if the icon even exists. If not use the caste's default sprite

Comment (harryob) [reply code/modules/mob/living/carbon/xenomorph/update_icons.dm:56]: okay but it doesn't compile with `(!icon_state) in icon_states(icon_xeno)`, `!(icon_state in icon_states(icon_xeno)`

Comment (aceroyuno) [reply code/modules/mob/living/carbon/xenomorph/update_icons.dm]: yeah, it should work now

**Comment by aceroyuno** [comment]
can anyone unstale this pr and unclose it? It's ready to merge.

**Pull request closed without merging**